qcengine/programs/adcc.py

```diff
@@ -0,0 +1,155 @@
+"""
+Calls adcc
+"""
+import json
+import os
+import sys
+from pathlib import Path
+import importlib
+from typing import TYPE_CHECKING, Any, Dict, Optional, Tuple
+import numpy as np
+
+from qcelemental.models import AtomicResult, AtomicResultProperties, Provenance
+from qcelemental.util import deserialize, parse_version, safe_version, which, which_import
+
+from ..exceptions import InputError, RandomError, ResourceError, UnknownError
+from .model import ProgramHarness
+
+if TYPE_CHECKING:
+    from qcelemental.models import AtomicInput
+
+    from ..config import TaskConfig
+
+
+def is_module_available(module):
+    """Check using importlib if a module is available."""
+    import importlib
+
+    try:
+        mod = importlib.import_module(module)
+    except ImportError:
+        return False
+    return True
+
+
+class AdccHarness(ProgramHarness):
+
+    _defaults = {
+        "name": "adcc",
+        "scratch": False,
+        "thread_safe": False,
+        "thread_parallel": True,
+        "node_parallel": False,
+        "managed_memory": True,
+    }
+    version_cache: Dict[str, str] = {}
+
+    class Config(ProgramHarness.Config):
+        pass
+
+    @staticmethod
+    def found(raise_error: bool = False) -> bool:
+        """Whether adcc harness is ready for operation.
+        Parameters
+        ----------
+        raise_error: bool
+            Passed on to control negative return between False and ModuleNotFoundError raised.
+        Returns
+        -------
+        bool
+            If adcc is found, returns True.
+            If raise_error is False and adcc is missing, returns False.
+            If raise_error is True and adcc is missing, the error message is raised.
+        """
+
+        adcc_found = is_module_available("adcc")
+        if adcc_found:
+            return True
+        if raise_error:
+            raise ModuleNotFoundError("adcc not found.")
+        return False
+
+    def get_version(self) -> str:
+        self.found(raise_error=True)
+        import adcc
+
+        # candidate_version = self.version_cache[which_prog]
+        candidate_version = adcc.__version__
+
+        return candidate_version
```

Comment:
https://github.com/MolSSI/QCEngine/blob/master/qcengine/programs/xtb.py#L48-L58 can be imitated directly to get the `version_cache`

Reply:
Done.